lib/app/app.rb

```diff
@@ -28,6 +29,8 @@ class ConnectorsWebApp < Sinatra::Base
   register Sinatra::ConfigFile
   config_file ConnectorsApp::CONFIG_FILE
 
+  DEFAULT_THREAD_COUNT = Concurrent.processor_count.freeze
```

Comment:
This is probably fine, but I'm noticing the _Concurrent_ documentation leaves this method undocumented 🤷

Reply:
Ah, that's interesting. The number of threads you can spin per *process* can be set much higher depending on what you do. Here, the app is I/O bound, so in the same process, you can have a lot of concurrent threads without being blocked by the GIL.  So `DEFAULT_THREAD_COUNT` can probably be set higher.

Python always had preemptive threads, but I've learned that this is something that's been added a bit later in CRuby 

https://bugfactory.io/blog/multithreading-in-the-mri-ruby-interpreter/

So if this still the current model i n `2.6.x` (no reason it's not) you have a pretty efficient context switching when there are dozains of threads running in the same process, as long as they stay I/O bound, which is our case ~ we call a backend through a socket and sit there waiting.

Reply:
Can we make this change (change to the config behaviour) a different PR, so that we can have a wider discussion about it?

Reply:
@artem-shelkovnikov do we have to? I mean, it's just the default. The number can be set to whatever in config. I just thought it would be a bit more convenient to have a reasonable default instead of crash the app on start when the value in config is unset, as it previously happened.